### ISSUE TITLE
@inject for classes

### ISSUE THREAD

**Issue by rmk135** [rmk135/dependency_injector#92]
Make `@inject` to work with classes. 

`@inject` should look for class's `__init__()` and apply injections to it. If decorated class has no ``__init__()``, raise an error.

```python
@di.inject(arg1=1)
@di.inject(arg2=2)
class A(object):

    def __init__(self, arg1, arg2):
        self.arg1 = arg1
        self.arg2 = arg2

# The same as:

class A(object):

    @di.inject(arg1=1)
    @di.inject(arg2=2)
    def __init__(self, arg1, arg2):
        self.arg1 = arg1
        self.arg2 = arg2
```

**Comment by rmk135** [comment]
Feature is implemented in 367beebf7293bc1bb4504f33c074834c57c2c428

**Comment by rmk135** [comment]
Added docs and tests, merged to master
